### PR TITLE
[lit-next] Nested directives whose parent returns `noChange` are unaffected

### LINKED ISSUE #1519: [lit-next] Ensure nested directives handle `noChange` as expected
When a directive returns `noChange` any nested directive value it might otherwise return should be unchanged. Currently, the `resolveDirective` helper does not handle `noChange` and treats this as a value that should clear the nested directive. The nested directive gets removed and then re-created if it is subsequently returned. This prevents, for example, the `guard` directive from being able to guard a nested directive from running.

To address this, `resolveDirective` can special case `noChange` to bail early and simply not process any nested directives.

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #1519. Now when a parent directive that previous returned another directive returns `noChange` the child directive is unaffected, meaning it does not update and is not disconnected or re-created.

### CODE REVIEW

File: packages/lit-html/src/test/directives/guard_test.ts [hunk 1]
```diff
@@ -163,4 +164,39 @@ suite('guard', () => {
 
     assert.equal(callCount, 2);
   });
+
+  test.only('guards directive from running', () => {
```
Comment (kevinpschaaf) [inline packages/lit-html/src/test/directives/guard_test.ts:168]: Remove the `only`

**Review by kevinpschaaf — APPROVED** [review]
LGTM, but need to remove `test.only` and make sure tests pass

File: packages/lit-html/src/test/directives/guard_test.ts (hunk 1 above)
Comment (kevinpschaaf) [reply packages/lit-html/src/test/directives/guard_test.ts:168]: Made a PR to catch these: https://github.com/Polymer/lit-html/pull/1521

**Review by justinfagnani — APPROVED** [review]

File: packages/lit-html/src/test/directives/guard_test.ts (hunk 1 above)
Comment (sorvell) [reply packages/lit-html/src/test/directives/guard_test.ts]: Fixed.

**Pull request merged**